### PR TITLE
DROOLS-2264 An option to enable strict conformance mode

### PR DESCRIPTION
**Pull request by kurobako**
https://issues.jboss.org/browse/DROOLS-2264

### CODE REVIEW

**Comment by kurobako** [comment]
jenkins retest this

File: kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/DMNAssemblerTest.java [hunk 1]
```diff
@@ -28,4 +39,48 @@ public void testDuplicateModel() {
...
+        DMNModel model = runtime.getModel( "http://www.trisotech.com/dmn/definitions/_ecf4ea54-2abc-4e2f-a101-4fe14e356a46", "strictMode" );
+        DMNContext ctx = runtime.newContext();
+        ctx.set( "timestring", "2016-12-20T14:30:22z" );
+        DMNResult result = runtime.evaluateAll(model, ctx);
+        assertEquals( DateTimeFormatter.ISO_TIME.parse( "14:30:22z", OffsetTime::from ), result.getDecisionResultByName( "time" ).getResult() );
+    }
+
+    @Test
+    public void testStrictMode() {
+        System.setProperty("org.kie.dmn.strictConformance", "true");
```
Comment (tarilabs) [inline kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/DMNAssemblerTest.java:55]: Just for completeness, I believe it is always a good practice to wrap this kind of statement in a try-catch, so to have the "clearing" in a finally block. This way we ensure the clean-up code is executed, without relying on potential different behavior from the test framework.

**Review by tarilabs — COMMENTED** [review]
thanks!

File: kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/DMNAssemblerTest.java (hunk 1 above)
Comment (kurobako) [reply kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/DMNAssemblerTest.java:55]: I dare to disagree -- these are special tools with clear purpose while try-finally adds noise to what test is actually doing. Though I think I should change annotation from @Before to @After: "All @After methods are guaranteed to run even if a Before or Test method throws an exception".

Comment (kurobako) [reply kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/DMNAssemblerTest.java:55]: updated pr, changed annotation from @before to @after

**Review by etirelli — APPROVED** [review]

**Pull request merged**